File: src/lib/icon-tones.ts
// Soft, editorial accent tones for icon "chips" — a small coloured circle/square
// behind a lucide icon. These mirror the palette used on the redesigned
// dashboard (sage, periwinkle, lavender, blush) plus the brand gold/teal.
//
// Usage: <span className={`... ${iconTones.sage}`}><Icon /></span>

export const iconTones = {
  sage: "bg-[#E6EDE4] text-[#6E8A6C]",
  periwinkle: "bg-[#E6E8F6] text-[#6E74B8]",
  lavender: "bg-[#EEE6F5] text-[#8A6CB0]",
  blush: "bg-[#F6E6EA] text-[#BC7488]",
  gold: "bg-gold/15 text-gold-dark",
  teal: "bg-teal/10 text-teal",
  blue: "bg-blue-50 text-blue-600",
  clay: "bg-clay-100 text-clay-500",
  emerald: "bg-emerald-50 text-emerald-600",
  amber: "bg-amber-50 text-amber-600",
  rose: "bg-rose-50 text-rose-500",
} as const;

export type IconTone = keyof typeof iconTones;

/** Resolve a tone name to its classes, defaulting to gold. */
export function toneClass(tone: IconTone = "gold"): string {
  return iconTones[tone];
}

/**
 * Soft glow colours (solid bg, used blurred at low opacity) that pair with each
 * tone — for the circular halos behind premium empty-state illustrations.
 */
export const toneGlow: Record<IconTone, string> = {
  sage: "bg-[#6E8A6C]",
  periwinkle: "bg-[#6E74B8]",
  lavender: "bg-[#8A6CB0]",
  blush: "bg-[#BC7488]",
  gold: "bg-gold",
  teal: "bg-teal",
  blue: "bg-blue-400",
  clay: "bg-clay-400",
  emerald: "bg-emerald-400",
  amber: "bg-amber-400",
  rose: "bg-rose-400",
};

File: mobile/lib/core/theme/icon_tones.dart
import 'package:flutter/material.dart';

import 'app_colors.dart';

/// Soft, editorial accent tones for icon "chips" — a small coloured square
/// behind an icon. Port of `src/lib/icon-tones.ts`; the background/foreground
/// pairs and the glow colours are the same values the web uses.
enum IconTone {
  sage,
  periwinkle,
  lavender,
  blush,
  gold,
  teal,
  blue,
  clay,
  emerald,
  amber,
  rose,
}

class ToneColors {
  const ToneColors({
    required this.background,
    required this.foreground,
    required this.glow,
  });

  final Color background;
  final Color foreground;

  /// Solid colour used blurred at low opacity behind empty-state art.
  final Color glow;
}

const Map<IconTone, ToneColors> iconTones = {
  IconTone.sage: ToneColors(
    background: Color(0xFFE6EDE4),
    foreground: Color(0xFF6E8A6C),
    glow: Color(0xFF6E8A6C),
  ),
  IconTone.periwinkle: ToneColors(
    background: Color(0xFFE6E8F6),
    foreground: Color(0xFF6E74B8),
    glow: Color(0xFF6E74B8),
  ),
  IconTone.lavender: ToneColors(
    background: Color(0xFFEEE6F5),
    foreground: Color(0xFF8A6CB0),
    glow: Color(0xFF8A6CB0),
  ),
  IconTone.blush: ToneColors(
    background: Color(0xFFF6E6EA),
    foreground: Color(0xFFBC7488),
    glow: Color(0xFFBC7488),
  ),
  // bg-gold/15, text-gold-dark
  IconTone.gold: ToneColors(
    background: Color(0x26C8963E),
    foreground: AppColors.goldDark,
    glow: AppColors.gold,
  ),
  // bg-teal/10, text-teal
  IconTone.teal: ToneColors(
    background: Color(0x1A4A9B8E),
    foreground: AppColors.teal,
    glow: AppColors.teal,
  ),
  IconTone.blue: ToneColors(
    background: AppColors.blue50,
    foreground: AppColors.blue600,
    glow: Color(0xFF60A5FA),
  ),
  IconTone.clay: ToneColors(
    background: AppColors.clay100,
    foreground: AppColors.clay500,
    glow: AppColors.clay400,
  ),
  IconTone.emerald: ToneColors(
    background: AppColors.emerald50,
    foreground: AppColors.emerald600,
    glow: Color(0xFF34D399),
  ),
  IconTone.amber: ToneColors(
    background: AppColors.amber50,
    foreground: AppColors.amber600,
    glow: Color(0xFFFBBF24),
  ),
  IconTone.rose: ToneColors(
    background: AppColors.rose50,
    foreground: AppColors.rose500,
    glow: Color(0xFFFB7185),
  ),
};

/// Resolve a tone to its colours, defaulting to gold.
ToneColors toneColors([IconTone tone = IconTone.gold]) =>
    iconTones[tone] ?? iconTones[IconTone.gold]!;
